contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,712 @@
+        if (!ownerCanModifySeries(seriesID)) {
+            revert AllArtistsRevokedOwnerRightsError(seriesID);
+        }
+        _validateArtistsNotRevoked(artistAddrs);
+
+        // Remove current artists
+        Series storage series = seriesDetails[seriesID];
+        for (uint256 i = 0; i < series.artistIDs.length; i++) {
+            uint256 artistID = series.artistIDs[i];
+            isArtistIDInSeries[seriesID][artistID] = false;
+            _removeSeriesFromArtist(artistID, seriesID);
+        }
+        delete series.artistIDs;
+
+        // Add new artists
+        _addArtistsToSeries(seriesID, artistAddrs);
+
+        emit SeriesUpdated(
+            seriesID,
+            series.artistIDs,
+            series.metadataURI,
+            series.contractTokenDataURI
+        );
+    }
+
+    // ============ Public/External Functions: Co-Artist Management ============
+
+    /**
+     * @dev Proposes a new co-artist for a series
+     */
+    function proposeCoArtist(
+        uint256 seriesID,
+        address proposedArtistAddr
+    ) external seriesExists(seriesID) onlyOwnerOrArtist(seriesID) {
+        if (proposedArtistAddr == address(0)) {
+            revert ZeroAddressNotAllowedError();
+        }
+        
+        _ensureArtistHasID(proposedArtistAddr);
+        uint256 proposedArtistID = addressToArtistID[proposedArtistAddr];
+
+        if (seriesPendingCoArtist[seriesID][proposedArtistID]) {
+            revert AlreadyProposedError(seriesID, proposedArtistAddr);
+        }
+
+        seriesPendingCoArtist[seriesID][proposedArtistID] = true;
+        _addPendingRequest(proposedArtistID, seriesID);
+
+        emit CoArtistProposed(seriesID, proposedArtistID);
+    }
+
+    /**
+     * @dev Cancels a co-artist proposal
+     */
+    function cancelProposeCoArtist(
+        uint256 seriesID,
+        address proposedArtistAddr
+    ) external seriesExists(seriesID) onlyOwnerOrArtist(seriesID) {
+        uint256 proposedArtistID = addressToArtistID[proposedArtistAddr];
+        if (!seriesPendingCoArtist[seriesID][proposedArtistID]) {
+            revert NoProposalExistsError(seriesID, proposedArtistAddr);
+        }
+
+        seriesPendingCoArtist[seriesID][proposedArtistID] = false;
+        _removePendingRequest(proposedArtistID, seriesID);
+
+        uint256 artistID = addressToArtistID[msg.sender];
+        emit CoArtistProposalCancelled(seriesID, artistID, proposedArtistID);
+    }
+
+    /**
+     * @dev Confirms participation as a co-artist
+     */
+    function confirmAsCoArtist(uint256 seriesID) external seriesExists(seriesID) {
+        uint256 artistID = addressToArtistID[msg.sender];
+        if (artistID == 0) {
+            revert NotAnArtistError(msg.sender);
+        }
+        if (!seriesPendingCoArtist[seriesID][artistID]) {
+            revert NotAPendingProposalError(seriesID, msg.sender);
+        }
+
+        Series storage series = seriesDetails[seriesID];
+        series.artistIDs.push(artistID);
```

Comment:
This logic could be presented already in `_addArtistsToSeries`, could it be reused?